capa/features/extractors/ts/query.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass
+
+from tree_sitter import Language
+from tree_sitter.binding import Query
+
+import capa.features.extractors.ts.build
+from capa.features.extractors.script import LANG_CS
+
+
+@dataclass
+class QueryBinding:
+    language: Language
+    new_object: Query
+    new_object_field_name: str
+    function_definition: Query
+    function_definition_field_name: str
+    function_call: Query
+    function_call_field_name: str
+    string_literal: Query
+    integer_literal: Query
+    namespace: Query
+    global_statement: Query
+
+    def __init__(self, language: str):
+        self.language = Language(capa.features.extractors.ts.build.build_dir, language)
+        if language == LANG_CS:
+            self.new_object = self.language.query("(object_creation_expression) @object.new")
+            self.new_object_field_name = "type"
+            self.function_definition = self.language.query("(local_function_statement) @function.definition")
+            self.function_definition_field_name = "name"
+            self.function_call = self.language.query("(invocation_expression) @function.call")
+            self.function_call_field_name = "function"
+            self.string_literal = self.language.query("(string_literal) @string-literal")
+            self.integer_literal = self.language.query("(integer_literal) @integer-literal")
+            self.namespace = self.language.query("(using_directive\n\t(qualified_name) @namespace)")
+            self.global_statement = self.language.query("(global_statement) @global-statement")
```

Comment:
this is ok, though i think we could make this even more declarative by putting the queries within separate files within a directory named after the language?

alternatively, maybe a file per language with a dict or dataclass literal containing the queries.

basically, there's a little line noise here with the assignment, method call, and query all together, which makes it more difficult to inspect the queries. we should try to make the queries as easy as possible for readers to inspect and understand.

Reply:
I refactored the queries into a global serialized JSON-like object that is optionally deserialized by a QueryBindingFactory when instantiating the QueryBinding object. Let me know if this fix is sufficient.